backend/curation/api/v1/curation/collections/common.py

```diff
@@ -49,50 +49,55 @@ def extract_doi_from_links(links: List[Link]) -> Tuple[Optional[str], List[dict]
-            # and datasets should expose the private url (based on version_id)
-            collection_id = collection_version.version_id
-            collection_url = f"{get_collections_base_url()}/collections/{collection_id.id}"
-            revision_of = collection_version.collection_id.id
-            use_canonical_url = False
-        else:
-            # If it's an unpublished, unrevised collection, then collection_url will point to the permalink
-            # (aka the link to the canonical_id) and the collection_id will point to version_id.
-            # Also, revision_of should be None, and the datasets should expose the canonical url
-            collection_id = collection_version.collection_id
-            collection_url = f"{get_collections_base_url()}/collections/{collection_id.id}"
+    # get collection attributes based on endpoint type and published status
+    if not reshape_for_version_endpoint:
+        if is_published:
+            # Published
+            collection_url = f"{get_collections_base_url()}/collections/{collection_version.collection_id.id}"
             revision_of = None
+            if not user_info or not user_info.is_user_owner_or_allowed(collection_version.owner):
+                _revising_in = None
+            else:
+                _revising_in = business_logic.get_unpublished_collection_version_from_canonical(
+                    collection_version.collection_id
+                )
+            revising_in = _revising_in.version_id.id if _revising_in else None
             use_canonical_url = True
+        else:
+            # Unpublished - need to determine if it's a revision or first time collection
+            # For that, we look at whether the canonical collection is published
+            is_revision = collection_version.canonical_collection.originally_published_at is not None
+            if is_revision:
+                # If it's a revision, both collection_id and collection_url need to point to the version_id,
+                # and datasets should expose the private url (based on version_id)
+                collection_url = f"{get_collections_base_url()}/collections/{collection_version.version_id.id}"
+                revision_of = collection_version.collection_id.id
+                use_canonical_url = False
+            else:
+                # If it's an unpublished, unrevised collection, then collection_url will point to the permalink
+                # (aka the link to the canonical_id) and the collection_id will point to version_id.
+                # Also, revision_of should be None, and the datasets should expose the canonical url
+                collection_url = f"{get_collections_base_url()}/collections/{collection_version.collection_id.id}"
+                revision_of = None
+                use_canonical_url = True
+            revising_in = None
+    else:
+        collection_url = f"{get_collections_base_url()}/collections/{collection_version.version_id.id}"
```

Comment:
@nayib-jose-gloria why would we return a url for a historical version that 
1) currently doesn't do anything, and 
2) we will soon change to redirect to the published version

perhaps `2)` is the desired behavior? Thoughts, @brianraymor ?  I'm not saying there's a good answer given our current UI behavior.

Reply:
I don't mind getting rid of it for this endpoint. I don't really think it makes sense in the context of querying for a particular version to get a URL out to the canonical collection, which may not match the version you're querying for. We also already provide the canonical collection_id in case you do want access to it.

tbh I linked out to the collection version url because we currently can access historical versions in the UI and I forgot we were going to make the change to redirect lol

Reply:
I'm fine with leaving it and accepting the impending behavior of redirect. But also would like to just get @brianraymor 's approval for this.

Reply:
I also vaguely remember us discussing wanting to have historical versions available in the UI at some point in the future, but we can update this in the future if that ever does happen, we don't need to future-proof it now